### ISSUE TITLE
expose the internal of BinaryCopyOutRow?

### ISSUE THREAD

**Issue by dovahcrow** [sfackler/rust-postgres#743]
Hi devs! We currently have a use case where we want to get the internal Bytes from a BinaryCopyOutRow instead of parsing types out, as well as its ranges. We use transmute now for getting the internal fields. Is there any possibilities to expose these fields in the future version?

**Comment by sfackler** [comment]
You should be able to make a FromSql implementation that accepts all types and just grabs the entire raw data buffer.

**Comment by dovahcrow** [comment]
@sfackler yes, this is a good solution in general. However, we are doing some zero-copy stuff so we'd like to store the Bytes of some fields of a row to someplace for later use, at the cost of just cloning the smart pointer of Bytes. The FromSql trait forbids us to do zero-copy because it passes in a &[u8].

**Comment by sfackler** [comment]
Do you want the Bytes for each individual value or the whole row?

**Comment by dovahcrow** [comment]
I think providing a function e.g. into_inner, to destruct the row into bytes and ranges would be sufficient. Users can decide how to handle the bytes.
